File: programme/pattern-1/pattern-1.c
// Write a program to print the given pattern
// * * * * *
//   * * * *
//     * * *
//       * *
//         *

#include <stdio.h>

#define PATTERN_STR ("*")

void pattern(int n)
{
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n - 1; j++)
        {
            if (j >= i)
            {
                printf("%s ", PATTERN_STR);
            }
            else
            {
                printf("  ");
            }
        }
        printf("%s\n", PATTERN_STR);
    }
}

int main(void)
{
    int n;
    printf("Input : ");
    scanf("%d", &n);
    pattern(n);
    return 0;
}

File: programme/pattern-1/pattern-1.js
// Write a program to print the given pattern
// * * * * *
//   * * * *
//     * * *
//       * *
//         *

"use strict";

// perform following command to run programme
// node pattern-1.js

const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

function pattern(n) {
  for (let i = 0; i < n; i++) {
    console.log(
      Array(n - i)
        .fill("*")
        .join(" ")
        .padStart(2 * n - 1)
    );
  }
}

rl.question("Input  : ", (n) => {
  console.log("Output :");
  n = parseInt(n);
  if (isNaN(n)) {
    console.error("Some error occured");
  } else {
    pattern(n);
  }
  rl.close();
});
